command/debug.go

```diff
@@ -0,0 +1,945 @@
+package command
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/api"
+	gatedwriter "github.com/hashicorp/vault/helper/gated-writer"
+	"github.com/hashicorp/vault/sdk/helper/logging"
+	"github.com/hashicorp/vault/sdk/helper/strutil"
+	"github.com/hashicorp/vault/sdk/version"
+	"github.com/mholt/archiver"
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+)
+
+const (
+	// debugIndexVersion is tracks the canonical version in the index file
+	// for compatibility with future format/layout changes on the bundle.
+	debugIndexVersion = 1
+
+	// debugMinInterval is the minimum acceptable interval capture value. This
+	// value applies to duration and all interval-related flags.
+	debugMinInterval = 5 * time.Second
+
+	// debugDurationGrace is the grace period added to duration to allow for
+	// "last frame" capture if the interval falls into the last duration time
+	// value. For instance, using default values, adding a grace duration lets
+	// the command capture 5 intervals (0, 30, 60, 90, and 120th second) before
+	// exiting.
+	debugDurationGrace = 1 * time.Second
+
+	// debugCompressionExt is the default compression extension used if
+	// compression is enabled.
+	debugCompressionExt = ".tar.gz"
+
+	// fileFriendlyTimeFormat is the time format used for file and directory
+	// naming.
+	fileFriendlyTimeFormat = "2006-01-02T15-04-05Z"
+)
+
+// debugIndex represents the data structure in the index file
+type debugIndex struct {
+	Version         int                    `json:"version"`
+	VaultAddress    string                 `json:"vault_address"`
+	ClientVersion   string                 `json:"client_version"`
+	Timestamp       time.Time              `json:"timestamp"`
+	Duration        int                    `json:"duration_seconds"`
```

Comment:
Why not make the field names like the json, i.e. append `Seconds` if they're a measure of seconds?